src/python_testing/matter_testing_infrastructure/chip/testing/matter_testing.py

```diff
@@ -1475,8 +1480,7 @@ async def attribute_guard(self, endpoint: int, attribute: ClusterObjects.Cluster
               if self.attribute_guard(condition2_needs_to_be_false_to_skip_step):
                   # skip step 2 if condition not met
            """
-        if self.stored_global_wildcard is None:
-            self.stored_global_wildcard = await self.global_wildcard
+        await self._populate_wildcard()
```

Comment:
Before self.global_wildcard is initialized only once, now, it is initialized for each await self._populate_wildcard(), why make this change?

Reply:
the None check is in the _populate_wildcard function, so this call does nothing if the value has already been stored. Moving this down means we don't get spurious warnings about x not awaited.